lib/widgets/message_list.dart

```diff
@@ -97,6 +97,31 @@ class MessageListAppBarTitle extends StatelessWidget {
   }
 }
 
+class ScrollToBottomButton extends StatelessWidget {
+  const ScrollToBottomButton({super.key, required this.scrollController, required this.visibleValue});
+  final ValueNotifier<bool> visibleValue;
+  final ScrollController scrollController;
+
+  Future<void> _navigateToBottom(BuildContext context) async {
+    scrollController.animateTo(0, duration: const Duration(milliseconds: 300), curve: Curves.easeIn);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return ValueListenableBuilder<bool>(
+      builder: (BuildContext context, bool value, Widget? child) {
+        return (value && child != null) ? child : const SizedBox.shrink();
+      },
+      valueListenable: visibleValue,
+      // TODO: fix hardcoded values for size and style here
+      child: IconButton(
+        tooltip: "Scroll to bottom",
+        icon: const Icon(Icons.expand_circle_down_rounded),
+        iconSize: 40,
+        style: IconButton.styleFrom(foregroundColor: const HSLColor.fromAHSL(0.5,240,0.96,0.68).toColor()),
```

Comment:
Hmm, oddly this line seems to have no effect for me. I get a transparent chevron in a black disc, with or without this line.

Reply:
Wacky. That screenshot was on Linux as well, on NixOS 22.11, but I was also running the PR on Android and seeing the same transparent-and-black there… and in general it'd be real odd if this behavior depended on the host OS.

A variable I'd guess is more likely to cause the difference is the Flutter version. That was on Flutter 3.12.0-14.0.pre.11, aka commit flutter/flutter@e0a9ad15b. … OK, indeed:
* On my Debian desktop (I was in the office yesterday, and at home today, so different machines), I reproduce the same transparent-and-black when I use that same Flutter version 3.12.0-14.0.pre.11, which is a couple of weeks old.
* But if I use a *current* Flutter version from main today — 3.13.0-7.0.pre.12 — then I get what I think you intended.

That raises the question of what change in Flutter is making that difference. In general we target the latest Flutter from main, so if they fixed a bug we don't need to worry about old versions… but I suspect this one was the flipping of the Material 3 switch by default:
https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/flutter.3A.20.60useMaterial3.60.20now.20.60true.60.20by.20default/near/1611211
OK, and when I turn that off:
```diff
--- a/lib/widgets/app.dart
+++ b/lib/widgets/app.dart
@@ -12,6 +12,7 @@ class ZulipApp extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     final theme = ThemeData(
+      useMaterial3: false,
```
it indeed goes back to transparent-and-black.

---

I think we are likely to apply that `useMaterial3: false` change soon, because there's a few things in the app that currently look odd in M3. We'll then eventually turn it back to true again, after fixing those. So let's find a way to make this button have the desired color both with `useMaterial3: false` and the new default of `useMaterial3: true`.

Reply:
(Filed #225 for that M3 migration.)

Reply:
Thanks for looking into the cause of the bug here. I was able to reproduce the same behavior on my end. I believe I've fixed it in 686d8cfd3d5225f01fd951d3f5e33a5ab2fa44be by using the `color` attribute rather than applying a `style`. It's more straightforward in any case.

I've attached a screenshot where I've applied `useMaterial3: false` and the color continues to render properly.

![zulip_flutter_with_blue_button](https://github.com/zulip/zulip-flutter/assets/98299/53234a98-0058-4464-a945-eb960e937caa)